icu4c/source/i18n/measunit_impl.h

```diff
@@ -330,6 +302,34 @@ struct U_I18N_API MeasureUnitImplWithIndex : public UMemory {
     }
 };
 
+// Export explicit template instantiations of MaybeStackArray, MemoryPool and
+// MaybeStackVector. This is required when building DLLs for Windows. (See
+// datefmt.h, collationiterator.h, erarules.h and others for similar examples.)
+#if U_PF_WINDOWS <= U_PLATFORM && U_PLATFORM <= U_PF_CYGWIN
+template class U_I18N_API MaybeStackArray<SingleUnitImpl *, 8>;
+template class U_I18N_API MemoryPool<SingleUnitImpl, 8>;
+template class U_I18N_API MaybeStackVector<SingleUnitImpl, 8>;
+template class U_I18N_API MaybeStackArray<MeasureUnitImplWithIndex *, 8>;
+template class U_I18N_API MemoryPool<MeasureUnitImplWithIndex, 8>;
+template class U_I18N_API MaybeStackVector<MeasureUnitImplWithIndex, 8>;
+#endif
```

Comment:
Can these be combined into a single ifdef?

Reply:
somehow the windows build is not happy about the PR. 
so on chromium, the template have to be define AFTER the whole class to build
but on here, the template have to be define BEFORE it got used to build
MaybeStackVector<MeasureUnitImplWithIndex, 8> is USED before it got defined so I don't know how to fix that part yet.

Reply:
Let's worry about about the combining the #ifdef after we can figure out how to make it build on BOTH places first. Because until then, I may need to move them around and it is much easier to keep them under different #ifdef to move around until where it will build first.

Reply:
if that is just too hard, I will just patch v8 locally and abandon this PR instead....

Reply:
DONE

Reply:
I see you had to move the `MeasureUnitImplWithIndex` as well. Thanks Frank!